packages/utah-design-system/src/components/Button.tsx

```diff
@@ -55,6 +59,17 @@ export function Button(props: ButtonProps) {
           className,
         }),
       )}
-    />
+    >
+      {composeRenderProps(props.children, (children) => (
+        <>
+          {children}
+          {props.isPending && (
+            <span className="ml-2">
+              <Spinner />
+            </span>
+          )}
```

Comment:
It might be nice to add a story to show this state.

Reply:
You can, at least in dev, change the is pending, disabled etc state. I can add an explicit story but didn't think it was necessary.

Reply:
I checked that this can be done in prod as well so I'm going to skip it. 
<img width="928" alt="image" src="https://github.com/user-attachments/assets/e0c45a8f-2ad4-4ca1-bc0e-7d2abb9e84fd" />